app/src/main/java/com/pajato/android/gamechat/common/InvitationManager.java

```diff
@@ -17,35 +17,185 @@
+            addedToCommRoomMemberList = false;
+        }
+
+        boolean isDone() {
+            return addedToAccountJoinList && addedToGroupMemberList && addedToCommRoomMemberList;
+        }
+    }
+
+    // Private constants.
+
+    private static final String APP_CODE = "aq5ca";
+    private static final String PLAY_STORE_LINK = "https://play.google.com/apps/testing/com.pajato.android.gamechat";
+    private static final String APP_PACKAGE_NAME = "com.pajato.android.gamechat";
+    private static final String WEB_LINK = "https://github.com/pajato/GameChat";
+
+    /** The logcat TAG. */
+    private static final String TAG = InvitationManager.class.getSimpleName();
+
+    // Private instance variables
+
+    /** The repository for any messages needed. */
+    private Map<Integer, String> messageMap = new HashMap<>();
+
+    /** Keep track of any outstanding invites to groups */
+    private Map<String, GroupInviteData> mInvitedGroups = new HashMap<>();
+
     // Public instance variables.
 
     /** The map managing group invitations. */
     private Map<String, List<String>> mGroupInviteMap = new HashMap<>();
 
     // Public instance methods.
 
+    /** Initialize the two central panels in the app: chat and game/activity. */
+    public void init(final AppCompatActivity context) {
+        messageMap.clear();
+        messageMap.put(R.string.HasJoinedMessage, context.getString(R.string.HasJoinedMessage));
+    }
+        /** Handle an account state change by updating the navigation drawer header. */
+    @Subscribe
+    public void onAuthenticationChange(final AuthenticationChangeEvent event) {
+        Account account = event != null ? event.account : null;
+        if (account == null) return;
+
+        // Update Firebase
+        boolean accountChanged = false;
+        for (String key : mInvitedGroups.keySet()) {
+            // If the account has already joined, don't add it again!
+            if (account.joinList.contains(key)) {
+                mInvitedGroups.remove(key);
+            } else {
+                account.joinList.add(key);
+                accountChanged = true;
+                mInvitedGroups.get(key).addedToAccountJoinList = true;
+                // Set up the watcher for the common room (the group watcher is set up by
+                // adding the group to the account joinList).
+                RoomManager.instance.setWatcher(key, mInvitedGroups.get(key).commonRoomKey);
+            }
+        }
+        if (accountChanged) {
+            AccountManager.instance.updateAccount(account);
+        }
+    }
+
+    /** Handle the room profile change */
+    @Subscribe public void onRoomProfileChange(@NonNull final ProfileRoomChangeEvent event) {
+        // If this room is one we care about, add the current account to the room memberIdList.
+        Account currAccount = AccountManager.instance.getCurrentAccount();
+        for (Map.Entry<String, GroupInviteData> entry : mInvitedGroups.entrySet()) {
```

Comment:
I've never really noticed the use of Map.Entry<> like this before.  I kinda like it.